query/excluder.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+//Excluder allows us to support different methods of excluding in the future
```

Comment:
I can make an issue for this, see if anyone picks it up, otherwise I'll gladly take care of it!